ebean-api/src/main/java/io/ebean/Query.java

```diff
@@ -1643,6 +1644,16 @@
    */
   String getGeneratedSql();
 
+  /**
+   * Execute the query with the given lock type and WAIT.
+   */
+  Query<T> withLock(LockType lockType);
+
+  /**
+   * Execute the query with the given lock type and lock wait.
+   */
+  Query<T> withLock(LockType lockType, LockWait lockWait);
+
   /**
    * Execute using "for update" clause which results in the DB locking the record.
```

Comment:
Maybe add to javadoc "same as calling withLock(UPDATE, WAIT)" ? Same for the other forUpdate() methods.